telemetry/data_window.go

```diff
@@ -80,6 +110,123 @@ var (
 	subWindowsLock    = sync.RWMutex{}
 )
 
+func getSQLSum(sqlTypeDeta *sqlType) int64 {
+	result := sqlTypeDeta.Use
+	result += sqlTypeDeta.Show
+	result += sqlTypeDeta.Begin
+	result += sqlTypeDeta.Commit
+	result += sqlTypeDeta.Rollback
+	result += sqlTypeDeta.Insert
+	result += sqlTypeDeta.Replace
+	result += sqlTypeDeta.Delete
+	result += sqlTypeDeta.Update
+	result += sqlTypeDeta.Select
+	result += sqlTypeDeta.Other
+	return result
+}
+
+func readSQLMetric(timepoint time.Time, SQLResult *sqlUsageData) error {
+	ctx := context.TODO()
+	promQL := "sum(tidb_executor_statement_total{}) by (instance,type)"
+	result, err := querySQLMetric(ctx, timepoint, promQL)
+	if err != nil {
+		if err1, ok := err.(*promv1.Error); ok {
+			return errors.Errorf("query metric error, msg: %v, detail: %v", err1.Msg, err1.Detail)
+		}
+		return errors.Errorf("query metric error: %v", err.Error())
+	}
+
+	anylisSQLUsage(result, SQLResult)
+	return nil
+}
+
+func querySQLMetric(ctx context.Context, queryTime time.Time, promQL string) (result pmodel.Value, err error) {
```

Comment:
Is it possible to use `queryMetrics()`(https://github.com/pingcap/tidb/blob/master/executor/metrics_reader.go#L90)?

Reply:
queryMetrics() returns Range vector but the func here returns Instant vector. We open an [#25163](https://github.com/pingcap/tidb/issues/25163) so that we can follow this and implement in the next Sprint.